libraries/GCS_MAVLink/MissionItemProtocol_Rally.cpp

```diff
@@ -71,6 +68,32 @@ MAV_MISSION_RESULT MissionItemProtocol_Rally::convert_MISSION_ITEM_INT_to_RallyL
         return MAV_MISSION_INVALID_PARAM7;
     }
     ret = {};
+
+    switch (cmd.frame) {
+        case MAV_FRAME_GLOBAL:
+        case MAV_FRAME_GLOBAL_INT:
+            ret.alt_frame = uint8_t(Location::AltFrame::ABSOLUTE);
+            break;
+
+        case MAV_FRAME_GLOBAL_RELATIVE_ALT:
+        case MAV_FRAME_GLOBAL_RELATIVE_ALT_INT:
+            ret.alt_frame = uint8_t(Location::AltFrame::ABOVE_HOME);
+            break;
+
+#if AP_TERRAIN_AVAILABLE
+        case MAV_FRAME_GLOBAL_TERRAIN_ALT:
+        case MAV_FRAME_GLOBAL_TERRAIN_ALT_INT:
+            ret.alt_frame = uint8_t(Location::AltFrame::ABOVE_TERRAIN);
+            break;
+#endif
+
+        default:
+            return MAV_MISSION_UNSUPPORTED_FRAME;
+    }
+
+    // Fresh points always use new alt frame format
+    ret.alt_frame_valid = true;
```

Comment:
If line 91 says the frame is unsupported, why is the alt_frame_valid still true?

Reply:
Line 91 is a return, the whole request is ignored.

Reply:
Ah, duh. Thanks! LGTM